src/ff/galois_field.rs

```diff
@@ -1,16 +1,48 @@
 use crate::{
-    ff::{GaloisField, Serializable},
-    secret_sharing::SharedValue,
+    ff::{Field, Serializable},
+    secret_sharing::{Block, SharedValue},
 };
-use bitvec::prelude::{BitArr, Lsb0};
+use bitvec::prelude::{bitarr, BitArr, Lsb0};
 use generic_array::GenericArray;
+use std::ops::Index;
 use typenum::{Unsigned, U1, U4, U5};
 
+/// Trait for data types storing arbitrary number of bits.
+pub trait GaloisField:
+    Field + Into<u128> + Index<usize, Output = bool> + Index<u32, Output = bool>
+{
+    const POLYNOMIAL: u128;
+
+    fn as_u128(self) -> u128 {
+        <Self as Into<u128>>::into(self)
+    }
+
+    /// Truncates the higher-order bits larger than `Self::BITS`, and converts
+    /// into this data type. This conversion is lossy. Callers are encouraged
+    /// to use `try_from` if the input is not known in advance.
+    fn truncate_from<T: Into<u128>>(v: T) -> Self;
+}
+
 // Bit store type definitions
 type U8_1 = BitArr!(for 8, in u8, Lsb0);
 type U8_4 = BitArr!(for 32, in u8, Lsb0);
 type U8_5 = BitArr!(for 40, in u8, Lsb0);
 
+impl Block for U8_1 {
+    type Size = U1;
+    const VALID_BIT_LENGTH: u32 = 8;
+}
+
+impl Block for U8_4 {
+    type Size = U4;
+    const VALID_BIT_LENGTH: u32 = 32;
+}
+
+impl Block for U8_5 {
+    type Size = U5;
+    const VALID_BIT_LENGTH: u32 = 40;
+}
+
```

Comment:
These are equivalent to former `Int` trait implementations, i.e., storage types.